File: AquaTrack/src/main/java/com/aquatrack/service/ManagerInvitationService.java
package com.aquatrack.service;

import com.aquatrack.dto.manager.CreateManagerInvitationRequestDto;
import com.aquatrack.dto.manager.ManagerActivationRequestDto;
import com.aquatrack.dto.manager.ManagerInvitationResponseDto;
import com.aquatrack.dto.manager.ManagerActivationDetailsResponseDto;

import java.util.List;

public interface ManagerInvitationService {

    // ==========================
    // Manager Invitation
    // ==========================

    /**
     * Creates a new manager invitation.
     *
     * @param requestDto Manager invitation request
     * @return Manager invitation response
     */
    ManagerInvitationResponseDto createManagerInvitation(
            CreateManagerInvitationRequestDto requestDto
    );

    // ==========================
    // Get Invitations
    // ==========================

    /**
     * Returns all manager invitations
     * of the logged-in Property Admin.
     *
     * @return List of manager invitations
     */
    List<ManagerInvitationResponseDto> getAllManagerInvitations();

    /**
     * Returns manager invitation by ID.
     *
     * @param invitationId Invitation ID
     * @return Manager invitation response
     */
    ManagerInvitationResponseDto getManagerInvitationById(
            Long invitationId
    );

    // ==========================================
    // Get Activation Details
    // ==========================================

    /**
     * Validates the invitation token
     * and returns manager invitation details.
     *
     * @param token Invitation token
     * @return Manager activation details
     */
    ManagerActivationDetailsResponseDto getActivationDetails(
            String token
    );

    // ==========================
    // Activation
    // ==========================

    /**
     * Activates manager account.
     *
     * @param requestDto Activation request
     */
    void activateManager(
            ManagerActivationRequestDto requestDto
    );



}

File: Aquatrack-frontend/src/services/managerInvitationService.js
import api from "../api/axios";

// ==========================================
// Get All Manager Invitations
// ==========================================

export async function getAllManagerInvitations() {

    const response = await api.get(
        "/manager-invitations"
    );

    return response.data;

}

// ==========================================
// Get Manager Invitation By Id
// ==========================================

export async function getManagerInvitationById(id) {

    const response = await api.get(
        `/manager-invitations/${id}`
    );

    return response.data;

}

// ==========================================
// Get Activation Details
// ==========================================

export async function getActivationDetails(token) {

    const response = await api.get(
        `/manager-invitations/activate?token=${token}`
    );

    return response.data;

}

// ==========================================
// Create Manager Invitation
// ==========================================

export async function createManagerInvitation(data) {

    const response = await api.post(
        "/manager-invitations",
        data
    );


    //  console.log("Manager List API:", response.data);

    return response.data;

}

// ==========================================
// Activate Manager
// ==========================================

export async function activateManager(data) {

    const response = await api.post(
        "/manager-invitations/activate",
        data
    );


    return response.data;

}